database-handler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+class DatabaseHandler
+{
+    private $table_name;
+
+    public function init()
+    {
+        global $wpdb;
+
+        $this->table_name = $wpdb->prefix . "lightning_publisher_payments";
+
+        $charset_collate = $wpdb->get_charset_collate();
+
+        $sql = "CREATE TABLE $this->table_name (
+        id bigint(20) NOT NULL AUTO_INCREMENT,
+        post_id bigint(20) NOT NULL,
+        payment_hash varchar(256) NOT NULL,
+        payment_request varchar(256) NOT NULL,
+        amount_in_satoshi double(30, 20) DEFAULT 0.0 NOT NULL,
+        exchange_rate double(30, 20) DEFAULT 0.0 NOT NULL,
+        exchange_currency varchar(10) NOT NULL,
+        created_at datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        settled_at datetime DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        state tinytext NOT NULL,
+        PRIMARY KEY  (id)
+        ) $charset_collate;";
+
+        require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+        dbDelta($sql);
+    }
+
+    public function store_invoice($post_id, $payment_hash, $payment_request, $amount, $currency)
+    {
+        global $wpdb;
+
+        // TODO: implement get exchange rate functionality
+        $exchange_rate = '';
```

Comment:
As this is the DatabaseHandler handler maybe it is better to pass in all the data including the exchange rate and then calculate the exchange rate in the wp-publisher.ph where we use this store_invoice function.